Add Vuetify, Tailwind, Styled Components, and Emotion recipes (#444)
storybookjs/frontpage: src/content/recipes/styled-components.md

<div class="aside aside__no-top">

This recipe assumes that you have a React app using Styled Components and have just set up Storybook >=6.0 using the [getting started guide](/docs/react/get-started/install). Don’t have this? Follow Tailwind's [installation instructions](https://styled-components.com/docs/basics#installation) then run:

```shell
# Add Storybook:
npx sb init
```

</div>

<RecipeHeader>

How to setup Styled Components and Storybook

</RecipeHeader>

> Styled Components is a popular UI framework for building React components with css-in-js, while Storybook is a tool for creating and testing UI components in isolation.
> This post will show you how to integrate these two tools to create a powerful and flexible development environment for building user interfaces with Styled Components.

Reviewer: ```suggestion
Styled Components is a popular library for building UI components with css-in-js, while Storybook is a tool for creating and testing UI components in isolation.
This post will show you how to integrate these two tools to create a powerful and flexible development environment for building user interfaces with Styled Components.
```

This post will explain how to:

> 2. 🧱 Use Styled Components in your components
> 3. 💅 Use a theme in your Stories
> 4. 🎨 Switch betweens themes in a click

Reviewer: ```suggestion
1. 🧱 Use Styled Components in your components
2. 💅 Use a theme in your Stories
3. 🎨 Switch betweens themes in a click
```

    secondary: '#1EA7FD',
    tertiary: '#DDDDDD',
    text: '#222222',
  },
  spacing: {
    padding: {
      small: 10,
      medium: 20,
      large: 30,
    },
    borderRadius: {
      small: 5,
      default: 10,
    },
  },
  typography: {
    type: {
      primary: '"Nunito Sans", "Helvetica Neue", Helvetica, Arial, sans-serif',
      code: '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
    },
    weight: {
      regular: '400',
      bold: '700',
      extrabold: '800',
      black: '900',
    },
    size: {
      s1: 12,
      s2: 14,
      s3: 16,
      m1: 20,
      m2: 24,
      m3: 28,
      l1: 32,
      l2: 40,
      l3: 48,
    },
  },
};
```

To share this theme with the components in Storybook, you'll need a [decorator](/docs/react/writing-stories/decorators).

Below I created a new file in `.storybook` called `withTheme.decorator.js` that will wrap your stories with Styled Component's `ThemeProvider`.

```js
// .storybook/withTheme.decorator.js

import { ThemeProvider } from 'styled-components';

export const withTheme = (Story) => (
  <ThemeProvider theme={lightTheme}>
    <Story />
  </ThemeProvider>
);
```

> All that is left to do is give this decorator to Storybook. Add the decorator to the decorator array in `.storybook/preview.js`:

Reviewer: ```suggestion
All that is left to do is give this decorator to Storybook. Add the decorator to the `decorators` array in `.storybook/preview.js`:
```

```js
// .storybook/preview.js

> import { withTheme } from 'styled-components';

Reviewer: ```suggestion
import { withTheme } from './withTheme.decorator';
```

<!-- prettier-ignore-start -->

<CodeSnippets
    paths={[
        'styled-components/Button.js.mdx',
        'styled-components/Header.js.mdx',
        'styled-components/Page.js.mdx',
    ]}
/>

<!-- prettier-ignore-end -->

## Add a theme switcher tool using `globalTypes`

Dark mode has become an increasingly popular offering on the web. This can be achieved quickly using themes.

![Completed Styled Components example with theme switcher](https://user-images.githubusercontent.com/18172605/208312563-875ca3b0-e7bc-4401-a445-4553b48068ed.gif)

Let's add the following dark theme to `theme.js`

```js
// ./src/theme.js

/* snipped for brevity */

export const darkTheme = {
  ...lightTheme,
  colors: {
    background: '#1b1c1d',
    backgroundInverse: '#333333',
    positive: '#9fd986',
    negative: '#df987d',
    primary: '#d43369',
    secondary: '#1b8bd0',
    tertiary: '#DDDDDD',
    text: '#FFFFFF',
  },
};
```

Now, to get the most out of your stories, there should be a way to toggle between themes in a click.

![Completed Styled Components example with theme switcher](https://user-images.githubusercontent.com/18172605/208312563-875ca3b0-e7bc-4401-a445-4553b48068ed.gif)

> To add the switcher, declare a [global variable](/docs/react/essentials/toolbars-and-globals) named `theme` in `.storybook/preview.js` and give it a list of supported themes to choose from.

Reviewer: ```suggestion
To add the switcher, declare a [global type](/docs/react/essentials/toolbars-and-globals) named `theme` in `.storybook/preview.js` and give it a list of supported themes to choose from.
```
